cmd/lncli/commands.go

```diff
@@ -1536,7 +1565,9 @@ func walletBalance(ctx *cli.Context) error {
 	defer cleanUp()
 
 	req := &lnrpc.WalletBalanceRequest{}
-	resp, err := client.WalletBalance(ctxb, req)
+	resp, err := handleSignals(ctxb, func(ctx context.Context) (proto.Message, error) {
```

Comment:
I think a better way of handling this would be to define a new method `getContext()` that returns a `context` that will be canceled in case of an interrupt. Passing `ctx` to `client.WalletBalance(ctx, req)` should then stop correctly.